runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -103,9 +103,15 @@
     public DynamicProtobufSerializer(CorfuRuntime corfuRuntime) {
         this.type = ProtobufSerializer.PROTOBUF_SERIALIZER_CODE;
 
-        // Create and register a protobuf serializer to read the table registry.
-        ISerializer protobufSerializer = createProtobufSerializer();
-        Serializers.registerSerializer(protobufSerializer);
+        // Create or get a protobuf serializer to read the table registry.
```

Comment:
Is it possible to have a test? It will validate your fix and prevent future regression.

Reply:
Good question. In order to reproduce the race condition, we need to run `new DynamicProtobufSerializer()` followed by a read operation which executes deserialize(), and make sure that the read returns before DynamicProtobufSerializer constructor completes. In our test environment it's hard to guarantee this condition since they are both lightning fast on local node. Do you have any thought?